webapp/src/js/services/android-api.js

```diff
@@ -171,6 +171,18 @@ angular.module('inboxServices').factory('AndroidApi',
           }
         },
 
+        /**
+         * Handle the response from the MRDT app
+         * @param response The stringified JSON response from the MRDT app.
+         */
+        mrdtTimeTakenResponse: function(response) {
+          try {
+            MRDT.respondTimeTaken(JSON.parse(response));
+          } catch(e) {
+            return $log.error(new Error('Unable to parse JSON response from android app: "' + response + '"'));
```

Comment:
Should you include `e` in this error message just incase this throws for some reason other than JSON parsing?